module/Finna/src/Finna/Recommend/Ontology.php

```diff
@@ -0,0 +1,512 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class Ontology implements RecommendInterface, TranslatorAwareInterface
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * Name of the cookie used to store the times shown total value.
+     */
+    public const COOKIE_NAME = 'ontologyRecommend';
+
+    public const TYPE_NONDESCRIPTOR = 'nondescriptor';
+
+    public const TYPE_SPECIFIER = 'specifier';
+
+    public const TYPE_HYPONYM = 'hyponym';
+
+    /**
+     * Finto connection class.
+     *
+     * @var Finto
+     */
+    protected $finto;
+
+    /**
+     * Cookie manager.
+     *
+     * @var CookieManager
+     */
+    protected $cookieManager;
+
+    /**
+     * Url helper.
+     *
+     * @var Url
+     */
+    protected $urlHelper;
+
+    /**
+     * Recommendation memory.
+     *
+     * @var RecommendationMemory
+     */
+    protected $recommendationMemory;
+
+    /**
+     * Raw configuration parameters.
+     *
+     * @var string
+     */
+    protected $rawParams;
+
+    /**
+     * Maximum API calls to make. A value of false indicates an unlimited number.
+     *
+     * @var int|bool
+     */
+    protected $maxApiCalls;
+
+    /**
+     * Maximum recommendations to show. A value of false indicates an unlimited
+     * number.
+     *
+     * @var int|bool
+     */
+    protected $maxRecommendations;
+
+    /**
+     * Maximum total for determining if the result set is small. A value of false
+     * indicates that all result sets should be considered small.
+     *
+     * @var int|bool
+     */
+    protected $maxSmallResultTotal;
+
+    /**
+     * Minimum total for determining if the result set is large. A value of false
+     * indicates that all result should be considered large.
+     *
+     * @var int|bool
+     */
+    protected $minLargeResultTotal;
+
+    /**
+     * Maximum number of times ontology recommendations can be shown per
+     * browser session. A value of false indicates an unlimited number.
+     *
+     * @var int|bool
+     */
+    protected $maxTimesShownPerSession;
+
+    /**
+     * Parameter object representing user request.
+     *
+     * @var \Laminas\StdLib\Parameters
+     */
+    protected $request;
+
+    /**
+     * Current search query.
+     *
+     * @var string
+     */
+    protected $lookfor;
+
+    /**
+     * Search results object.
+     *
+     * @var \VuFind\Search\Base\Results
+     */
+    protected $results;
+
+    /**
+     * Search ID.
+     *
+     * @var int
+     */
+    protected $searchId;
+
+    /**
+     * Total count of records in the result set.
+     *
+     * @var int
+     */
+    protected $resultTotal;
+
+    /**
+     * Ontology results.
+     *
+     * @var array
+     */
+    protected $ontologyResults;
+
+    /**
+     * Total number of API calls made.
+     *
+     * @var int
```

Comment:
Jos nämä voi asettaa falseksi, niin sen pitäisi näkyä tässäkin.

Reply:
`$maxApiCalls`:in voi asettaa falseksi, tämä on vain laskuri.